hive-metastore/src/test/java/org/apache/iceberg/hive/TestHiveCatalog.java

```diff
@@ -328,6 +329,140 @@ public void testCreateTableCustomSortOrder() throws Exception {
+        DB_NAME,
+        "tbl_alter_owner_noop_3",
+        ImmutableMap.of(),
+        ImmutableMap.of(),
+        UserGroupInformation.getCurrentUser().getUserName(),
+        UserGroupInformation.getCurrentUser().getUserName());
+  }
+
+  private void alterTableAndVerifyOwner(
+      String db,
+      String tbl,
+      Map<String, String> properties,
+      Map<String, String> updates,
+      String expectedOwnerPostCreate,
+      String expectedOwnerPostAlter)
+      throws IOException, TException {
+    Schema schema = getTestSchema();
+    PartitionSpec spec = PartitionSpec.builderFor(schema).bucket("data", 16).build();
+    TableIdentifier tableIdent = TableIdentifier.of(db, tbl);
+    String location = temp.newFolder(tbl).toString();
+    try {
+      Table table = catalog.createTable(tableIdent, schema, spec, location, properties);
+      org.apache.hadoop.hive.metastore.api.Table hmsTable = metastoreClient.getTable(db, tbl);
+      Assert.assertEquals(expectedOwnerPostCreate, hmsTable.getOwner());
+      Assert.assertFalse(hmsTable.getParameters().containsKey(HiveCatalog.HMS_TABLE_OWNER));
+      UpdateProperties updateOps = table.updateProperties();
+      updates.forEach(updateOps::set);
+      updateOps.commit();
+      hmsTable = metastoreClient.getTable(db, tbl);
+      Assert.assertEquals(expectedOwnerPostAlter, hmsTable.getOwner());
+      Assert.assertFalse(hmsTable.getParameters().containsKey(HiveCatalog.HMS_TABLE_OWNER));
+    } finally {
+      catalog.dropTable(tableIdent);
+    }
+  }
+
+  @Test
+  public void testRemoveTableOwner() throws IOException, TException {
+    removeTableOwnerAndVerify(
+        DB_NAME,
+        "tbl_remove_owner_1",
+        ImmutableMap.of(HiveCatalog.HMS_TABLE_OWNER, "some_owner"),
+        ImmutableSet.of(HiveCatalog.HMS_TABLE_OWNER),
+        "some_owner",
+        UserGroupInformation.getCurrentUser().getUserName());
+    removeTableOwnerAndVerify(
+        DB_NAME,
+        "tbl_remove_owner_noop_1",
+        ImmutableMap.of(),
+        ImmutableSet.of(HiveCatalog.HMS_TABLE_OWNER),
+        UserGroupInformation.getCurrentUser().getUserName(),
+        UserGroupInformation.getCurrentUser().getUserName());
+    removeTableOwnerAndVerify(
+        DB_NAME,
+        "tbl_remove_owner_noop_2",
+        ImmutableMap.of(HiveCatalog.HMS_TABLE_OWNER, "some_owner"),
+        ImmutableSet.of(),
+        "some_owner",
+        "some_owner");
+    removeTableOwnerAndVerify(
+        DB_NAME,
+        "tbl_remove_owner_noop_3",
+        ImmutableMap.of(HiveCatalog.HMS_TABLE_OWNER, "some_owner"),
+        ImmutableSet.of("unrelated_prop"),
+        "some_owner",
+        "some_owner");
+    removeTableOwnerAndVerify(
+        DB_NAME,
+        "tbl_remove_owner_noop_4",
+        ImmutableMap.of(),
+        ImmutableSet.of(),
+        UserGroupInformation.getCurrentUser().getUserName(),
+        UserGroupInformation.getCurrentUser().getUserName());
+  }
+
+  private void removeTableOwnerAndVerify(
```

Comment:
nit: this seems almost identical to alterTableAndVerifyOwner except 1 line that decides if we are altering or removing. Can you please merge these two function together and make the updateProperties operation pluggable or being decided by some parameter?

Reply:
Ack. thx